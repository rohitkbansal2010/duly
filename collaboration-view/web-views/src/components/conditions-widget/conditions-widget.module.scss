@import "~@styles/colors.scss";
@import "~@styles/main.scss";
@import "~@styles/typography.scss";
@import "~@styles/spacing-constants.scss";

@mixin common-title {
  margin: 1.75rem 0;
  font-weight: 800;
  font-size: var(--font-size-24px-4px);
  line-height: var(--line-height-26px-4px);
  @include nexa-heavy-highlight;
}

.conditionsWidget {
  display: flex;
  gap: 1rem;
  align-items: stretch;
  width: 100%;
  height: 100%;
  max-height: $widget-height;

  &Col {
    display: flex;
    flex-direction: column;
    width: 50%;

    &Title {
      @include common-title;
      color: $dupage-light-blue;
    }

    &CurrentTitle {
      @include common-title;
      color: $dupage-magenta;
    }

    &Content {
      flex-grow: 1;
      min-height: 0;
    }

    &Data {
      display: flex;
      flex-direction: column;
      gap: 1.75rem;
    }
  }

  &Skeleton {
    position: relative;
    height: 100%;
    background: $white;
    border-radius: 0.5rem;
  }
}
