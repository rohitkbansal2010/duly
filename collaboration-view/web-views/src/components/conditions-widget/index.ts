export { ConditionsWidget } from './conditions-widget';
