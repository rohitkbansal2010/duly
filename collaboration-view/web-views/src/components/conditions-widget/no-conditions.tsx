import { WidgetNoData } from '@components/widget-no-data';
import { conditionsIcon, currentConditionsIcon } from '@icons';

type NoConditionsWidgetProps = {
  isCurrent: boolean;
};

export const NoConditionsWidget = ({ isCurrent }: NoConditionsWidgetProps) =>
  (
    <WidgetNoData
      view={isCurrent ? 'magenta' : 'blue'}
      icon={<img src={isCurrent ? currentConditionsIcon : conditionsIcon} alt="no conditions" />}
      title={`No ${isCurrent ? 'Current' : 'Previous'} Health Conditions`}
      align="top"
    />
  );
