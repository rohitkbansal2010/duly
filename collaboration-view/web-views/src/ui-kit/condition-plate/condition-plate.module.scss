@import "~@styles/colors.scss";

@mixin common-icon {
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  width: 3.75rem;
  height: 3.75rem;
  padding: 0.75rem;
  border-radius: 100%;

  img {
    width: 100%;
    height: 100%;
  }
}

.conditionPlate {
  display: flex;
  align-items: center;
  padding: 1.25rem;
  background: white;
  border-radius: 0.5rem;

  &Icon {
    @include common-icon;
    flex-shrink: 0;
    background: lighten($dupage-light-blue, 40%);

    &Current {
      @include common-icon;
      background: lighten($dupage-magenta, 40%);
    }
  }

  &Data {
    margin-left: 1.25rem;
    word-break: break-word;

    &Name {
      margin: 0;
      font-weight: 800;
      font-size: var(--font-size-24px-4px);
      // line-height: 1.5rem;
    }

    &Date {
      font-size: var(--font-size-16px-4px);
      line-height: var(--line-height-24px-4px);
    }
  }
}
