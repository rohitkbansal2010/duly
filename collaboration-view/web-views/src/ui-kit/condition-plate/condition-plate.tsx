import { conditionsIcon, currentConditionsIcon } from '@icons';
import { formatMDYYYYDate } from '@utils';

import styles from './condition-plate.module.scss';

export type ConditionPlateProps = {
  title: string;
  date?: string;
  isCurrent?: boolean;
};

export const ConditionPlate = ({ title, date, isCurrent }: ConditionPlateProps) => 
  (
    <div className={styles.conditionPlate}>
      <div className={styles[`conditionPlateIcon${isCurrent ? 'Current' : ''}`]}>
        <img
          src={isCurrent ? currentConditionsIcon : conditionsIcon}
          alt="conditions icon"
        />
      </div>
      <div className={styles.conditionPlateData}>
        <h2 className={styles.conditionPlateDataName}>{title}</h2>
        <span className={styles.conditionPlateDataDate}>{date && formatMDYYYYDate(date)}</span>
      </div>
    </div>
  );
