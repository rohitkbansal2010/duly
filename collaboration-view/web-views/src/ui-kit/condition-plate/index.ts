export { ConditionPlate } from './condition-plate';
