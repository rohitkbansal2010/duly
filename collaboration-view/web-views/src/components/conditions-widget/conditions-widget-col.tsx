import { FadedScroll } from '@components/faded-scroll-2';
import { ConditionPlate } from '@ui-kit/condition-plate';

import styles from './conditions-widget.module.scss';
import { NoConditionsWidget } from './no-conditions';

export type ConditionType = {
  title: string;
  date?: string;
};

type ConditionsWidgetColProps = {
  isCurrent: boolean;
  title: string;
  conditions: ConditionType[];
};

export const ConditionsWidgetCol = ({
  isCurrent,
  title,
  conditions = [],
}: ConditionsWidgetColProps) =>
  (
    <div className={styles.conditionsWidgetCol}>
      <div>
        <h2 className={styles[isCurrent ? 'conditionsWidgetColCurrentTitle' : 'conditionsWidgetColTitle']}>
          {title}
        </h2>
      </div>
      <div className={styles.conditionsWidgetColContent}>
        {conditions.length
        ? (

          <FadedScroll height="100%">
            <div className={styles.conditionsWidgetColData}>
              {conditions.map(({ date, title: plateTitle }) =>
                  (
                    <ConditionPlate
                      isCurrent={isCurrent}
                      key={plateTitle}
                      title={plateTitle}
                      date={date}
                    />
                  ))}
            </div>
          </FadedScroll>
        )
        : <NoConditionsWidget isCurrent={isCurrent} />
      }
      </div>
    </div>
  );


