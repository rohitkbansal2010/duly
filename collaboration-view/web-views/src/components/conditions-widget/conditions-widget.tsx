import sortBy from 'lodash/sortBy';
import { useMemo } from 'react';
import { useSelector } from 'react-redux';

import { DulyLoader } from '@components/duly-loader';
import { HEALTH_CONDITIONS_LOADER_WIDTH } from '@constants';
import { AppointmentModuleWidgetsAlias } from '@enums';
import { useWidgetSkeleton } from '@hooks';
import { RootState } from '@redux/reducers';

import { ConditionsWidgetCol, ConditionType } from './conditions-widget-col';
import styles from './conditions-widget.module.scss';


const conditionSorter = (conditions: ConditionType[]) =>
  sortBy(
    conditions,
    (condition: ConditionType) =>
      (condition.date ? -new Date(condition.date) : 1),
    [ 'title' ]
  );

export const ConditionsWidget = () => {
  const conditions = useSelector(({ OVERVIEW_WIDGETS }: RootState) =>
    OVERVIEW_WIDGETS.conditions);

  const isSkeletonShown = useWidgetSkeleton(AppointmentModuleWidgetsAlias.CONDITIONS);

  const prevConditions = useMemo(() =>
    conditionSorter(
      conditions?.previousHealthConditions || []
    ), [ conditions?.previousHealthConditions ]);

  const currentConditions = useMemo(() =>
    conditionSorter(
      conditions?.currentHealthConditions || []
    ), [ conditions?.currentHealthConditions ]);

  if (isSkeletonShown) {
    return (
      <div className={styles.conditionsWidgetSkeleton}>
        <DulyLoader width={HEALTH_CONDITIONS_LOADER_WIDTH} />
      </div>
    );
  }

  return (
    <div className={styles.conditionsWidget}>
      <ConditionsWidgetCol
        conditions={currentConditions}
        isCurrent
        title="Current Health Conditions"
      />
      <ConditionsWidgetCol
        conditions={prevConditions}
        isCurrent={false}
        title="Previous Health Conditions"
      />
    </div>
  );
};
