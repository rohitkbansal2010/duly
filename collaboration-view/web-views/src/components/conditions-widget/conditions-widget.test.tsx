import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as redux from 'react-redux';

import { healthConditionsMockData } from '@mock-data';
import { ConditionPlate } from '@ui-kit/condition-plate';

import { ConditionsWidget } from './conditions-widget';
import { NoConditionsWidget } from './no-conditions';

const { previousHealthConditions, currentHealthConditions } = healthConditionsMockData;
const commonCount = previousHealthConditions.length + currentHealthConditions.length;

describe('HealthConditionsWidget', () => {
  let component: ReturnType<typeof mount | typeof shallow>;

  beforeEach(() => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((selector: (state: unknown) => unknown) =>
        selector({ OVERVIEW_WIDGETS: { conditions: healthConditionsMockData } }));

    component = mount(<ConditionsWidget />);
  });

  it('component should match to snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('no conditions widget should match to snapshot', () => {
    component = shallow(<NoConditionsWidget isCurrent={true}/>);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should contain right count of conditions', () => {
    expect(component.find(ConditionPlate)).toHaveLength(commonCount);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should contain render no conditions widget if there is no conditions', () => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((selector: (state: unknown) => unknown) =>
        selector({
          OVERVIEW_WIDGETS: {
            conditions: {
              previousHealthConditions: [],
              currentHealthConditions: [],
            },
          },
        }));

    component = mount(<ConditionsWidget />);

    expect(component.find(ConditionPlate)).toHaveLength(0);
    expect(component.find(NoConditionsWidget)).toHaveLength(2);
  });
});
