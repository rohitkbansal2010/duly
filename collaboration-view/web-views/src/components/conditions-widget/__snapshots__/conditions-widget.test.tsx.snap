// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthConditionsWidget component should contain right count of conditions 1`] = `
<ConditionsWidget>
  <div
    className="conditionsWidget"
  >
    <ConditionsWidgetCol
      conditions={
        Array [
          Object {
            "date": "2022-02-16T00:00:00Z",
            "title": "Pneumonia",
          },
          Object {
            "date": "2019-05-28T00:00:00Z",
            "title": "Chronic sinusitis",
          },
          Object {
            "title": "Diabetes mellitus, type II",
          },
          Object {
            "title": "Osteoarthritis",
          },
        ]
      }
      isCurrent={true}
      title="Current Health Conditions"
    >
      <div
        className="conditionsWidgetCol"
      >
        <div>
          <h2
            className="conditionsWidgetColCurrentTitle"
          >
            Current Health Conditions
          </h2>
        </div>
        <div
          className="conditionsWidgetColContent"
        >
          <FadedScroll
            height="100%"
          >
            <div
              className="scrollWrapper"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <div
                className="topScrollFade top0 default"
              />
              <div
                className="scrollContent"
              >
                <div
                  className="conditionsWidgetColData"
                >
                  <ConditionPlate
                    date="2022-02-16T00:00:00Z"
                    isCurrent={true}
                    key="Pneumonia"
                    title="Pneumonia"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Pneumonia
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          2/16/2022
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    date="2019-05-28T00:00:00Z"
                    isCurrent={true}
                    key="Chronic sinusitis"
                    title="Chronic sinusitis"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Chronic sinusitis
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          5/28/2019
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={true}
                    key="Diabetes mellitus, type II"
                    title="Diabetes mellitus, type II"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Diabetes mellitus, type II
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={true}
                    key="Osteoarthritis"
                    title="Osteoarthritis"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Osteoarthritis
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                </div>
              </div>
              <div
                className="bottomScrollFade bottom0 default"
              />
            </div>
          </FadedScroll>
        </div>
      </div>
    </ConditionsWidgetCol>
    <ConditionsWidgetCol
      conditions={
        Array [
          Object {
            "date": "2022-02-11T00:00:00Z",
            "title": "Pneumonia",
          },
          Object {
            "date": "2022-02-01T00:00:00Z",
            "title": "Dehydration",
          },
          Object {
            "date": "2021-12-23T00:00:00Z",
            "title": "Myocardial infarction (STEMI)",
          },
          Object {
            "title": "Hypertension associated with diabetes (HCC)",
          },
          Object {
            "title": "Kidney stones",
          },
        ]
      }
      isCurrent={false}
      title="Previous Health Conditions"
    >
      <div
        className="conditionsWidgetCol"
      >
        <div>
          <h2
            className="conditionsWidgetColTitle"
          >
            Previous Health Conditions
          </h2>
        </div>
        <div
          className="conditionsWidgetColContent"
        >
          <FadedScroll
            height="100%"
          >
            <div
              className="scrollWrapper"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <div
                className="topScrollFade top0 default"
              />
              <div
                className="scrollContent"
              >
                <div
                  className="conditionsWidgetColData"
                >
                  <ConditionPlate
                    date="2022-02-11T00:00:00Z"
                    isCurrent={false}
                    key="Pneumonia"
                    title="Pneumonia"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Pneumonia
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          2/11/2022
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    date="2022-02-01T00:00:00Z"
                    isCurrent={false}
                    key="Dehydration"
                    title="Dehydration"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Dehydration
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          2/1/2022
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    date="2021-12-23T00:00:00Z"
                    isCurrent={false}
                    key="Myocardial infarction (STEMI)"
                    title="Myocardial infarction (STEMI)"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Myocardial infarction (STEMI)
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          12/23/2021
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={false}
                    key="Hypertension associated with diabetes (HCC)"
                    title="Hypertension associated with diabetes (HCC)"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Hypertension associated with diabetes (HCC)
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={false}
                    key="Kidney stones"
                    title="Kidney stones"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Kidney stones
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                </div>
              </div>
              <div
                className="bottomScrollFade bottom0 default"
              />
            </div>
          </FadedScroll>
        </div>
      </div>
    </ConditionsWidgetCol>
  </div>
</ConditionsWidget>
`;

exports[`HealthConditionsWidget component should match to snapshot 1`] = `
<ConditionsWidget>
  <div
    className="conditionsWidget"
  >
    <ConditionsWidgetCol
      conditions={
        Array [
          Object {
            "date": "2022-02-16T00:00:00Z",
            "title": "Pneumonia",
          },
          Object {
            "date": "2019-05-28T00:00:00Z",
            "title": "Chronic sinusitis",
          },
          Object {
            "title": "Diabetes mellitus, type II",
          },
          Object {
            "title": "Osteoarthritis",
          },
        ]
      }
      isCurrent={true}
      title="Current Health Conditions"
    >
      <div
        className="conditionsWidgetCol"
      >
        <div>
          <h2
            className="conditionsWidgetColCurrentTitle"
          >
            Current Health Conditions
          </h2>
        </div>
        <div
          className="conditionsWidgetColContent"
        >
          <FadedScroll
            height="100%"
          >
            <div
              className="scrollWrapper"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <div
                className="topScrollFade top0 default"
              />
              <div
                className="scrollContent"
              >
                <div
                  className="conditionsWidgetColData"
                >
                  <ConditionPlate
                    date="2022-02-16T00:00:00Z"
                    isCurrent={true}
                    key="Pneumonia"
                    title="Pneumonia"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Pneumonia
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          2/16/2022
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    date="2019-05-28T00:00:00Z"
                    isCurrent={true}
                    key="Chronic sinusitis"
                    title="Chronic sinusitis"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Chronic sinusitis
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          5/28/2019
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={true}
                    key="Diabetes mellitus, type II"
                    title="Diabetes mellitus, type II"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Diabetes mellitus, type II
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={true}
                    key="Osteoarthritis"
                    title="Osteoarthritis"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIconCurrent"
                      >
                        <img
                          alt="conditions icon"
                          src="current-conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Osteoarthritis
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                </div>
              </div>
              <div
                className="bottomScrollFade bottom0 default"
              />
            </div>
          </FadedScroll>
        </div>
      </div>
    </ConditionsWidgetCol>
    <ConditionsWidgetCol
      conditions={
        Array [
          Object {
            "date": "2022-02-11T00:00:00Z",
            "title": "Pneumonia",
          },
          Object {
            "date": "2022-02-01T00:00:00Z",
            "title": "Dehydration",
          },
          Object {
            "date": "2021-12-23T00:00:00Z",
            "title": "Myocardial infarction (STEMI)",
          },
          Object {
            "title": "Hypertension associated with diabetes (HCC)",
          },
          Object {
            "title": "Kidney stones",
          },
        ]
      }
      isCurrent={false}
      title="Previous Health Conditions"
    >
      <div
        className="conditionsWidgetCol"
      >
        <div>
          <h2
            className="conditionsWidgetColTitle"
          >
            Previous Health Conditions
          </h2>
        </div>
        <div
          className="conditionsWidgetColContent"
        >
          <FadedScroll
            height="100%"
          >
            <div
              className="scrollWrapper"
              style={
                Object {
                  "height": "100%",
                }
              }
            >
              <div
                className="topScrollFade top0 default"
              />
              <div
                className="scrollContent"
              >
                <div
                  className="conditionsWidgetColData"
                >
                  <ConditionPlate
                    date="2022-02-11T00:00:00Z"
                    isCurrent={false}
                    key="Pneumonia"
                    title="Pneumonia"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Pneumonia
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          2/11/2022
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    date="2022-02-01T00:00:00Z"
                    isCurrent={false}
                    key="Dehydration"
                    title="Dehydration"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Dehydration
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          2/1/2022
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    date="2021-12-23T00:00:00Z"
                    isCurrent={false}
                    key="Myocardial infarction (STEMI)"
                    title="Myocardial infarction (STEMI)"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Myocardial infarction (STEMI)
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        >
                          12/23/2021
                        </span>
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={false}
                    key="Hypertension associated with diabetes (HCC)"
                    title="Hypertension associated with diabetes (HCC)"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Hypertension associated with diabetes (HCC)
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                  <ConditionPlate
                    isCurrent={false}
                    key="Kidney stones"
                    title="Kidney stones"
                  >
                    <div
                      className="conditionPlate"
                    >
                      <div
                        className="conditionPlateIcon"
                      >
                        <img
                          alt="conditions icon"
                          src="conditions-icon.svg"
                        />
                      </div>
                      <div
                        className="conditionPlateData"
                      >
                        <h2
                          className="conditionPlateDataName"
                        >
                          Kidney stones
                        </h2>
                        <span
                          className="conditionPlateDataDate"
                        />
                      </div>
                    </div>
                  </ConditionPlate>
                </div>
              </div>
              <div
                className="bottomScrollFade bottom0 default"
              />
            </div>
          </FadedScroll>
        </div>
      </div>
    </ConditionsWidgetCol>
  </div>
</ConditionsWidget>
`;

exports[`HealthConditionsWidget no conditions widget should match to snapshot 1`] = `
<WidgetNoData
  align="top"
  icon={
    <img
      alt="no conditions"
      src="current-conditions-icon.svg"
    />
  }
  title="No Current Health Conditions"
  view="magenta"
/>
`;
